src/meili/match_route.cc

```diff
@@ -141,15 +141,21 @@ void cut_segments(const std::vector<MatchResult>& match_results,
     if (!curr_match.is_break_point && curr_idx != last_idx) {
       continue;
     }
-    // we want to handle to loop by locating the correct target edge by comparing the distance alone
-    bool loop = prev_match.edgeid == curr_match.edgeid &&
-                prev_match.distance_along > curr_match.distance_along;
+    // we want to handle to loop by locating the correct target edge by comparing the distance alone.
+    // Note: there should be a proper tolerance check to see if the 'percent_along's are close
+    // to one another, but comparing percentages isn't geometrically correct. Some refactoring
+    // would be required if we want to strenghen this condition.
+    bool loop =
+        prev_match.edgeid == curr_match.edgeid && prev_match.percent_along > curr_match.percent_along;
```

Comment:
i dont understand the comment about correctness and i dont understand the significance of the change. can you elaborate on what this fixes?

Reply:
right, the rename.

Reply:
i tried to explain a bit below but this check is important. its possible to have a sequence of match points such that the match can go around the block and come back to the same edge but behind the previous match point. that is completely valid. this bit of code detects that and makes sure not to again split the same edge but rather find the next occurance of that edge in the sequence of matched edges.